mesonbuild/backend/backends.py

```diff
@@ -782,15 +770,22 @@ def get_regen_filelist(self):
         return deps
 
     def exe_object_to_cmd_array(self, exe):
-        if self.environment.is_cross_build() and \
-           isinstance(exe, build.BuildTarget) and exe.is_cross:
-            if self.environment.exe_wrapper is None and self.environment.need_exe_wrapper():
-                s = textwrap.dedent('''
-                    Can not use target {} as a generator because it is cross-built
-                    and no exe wrapper is defined or needs_exe_wrapper is true.
-                    You might want to set it to native instead.'''.format(exe.name))
-                raise MesonException(s)
         if isinstance(exe, build.BuildTarget):
+            if exe.for_machine > MachineChoice.BUILD:
```

Comment:
I know you can use `>` here, but I can't forsee ever adding another value to `MachineChoice`, and I think `if exe.for_machine is not MachineChoice.BUILD` is clearer